server.go

```diff
@@ -12,6 +12,12 @@ import (
 	"github.com/julienschmidt/httprouter"
 )
 
+// A ResponseWriter is an interface for types that can be marshalled into an
+// HTTP response body.
+type ResponseWriter interface {
+	WriteResponse(http.ResponseWriter)
+}
```

Comment:
I am suspicious of this. If you need to write something other than JSON, you can do that by writing to `jc.ResponseWriter` directly. I can see the appeal of "everything goes through `Encode`, but in practice we *want* non-standard behavior to stand out.

Reply:
Yeah, there aren't very many places that this would be used at least off the top of my head.  For aliased types used for Prometheus responses we use a helper anyways.

Reply:
I did not mean to commit that 😂